### ISSUE TITLE
Notification System using Web Sockets (Action Cable)

### ISSUE THREAD

**Issue by namangupta01** [publiclab/plots2#5742]
For Notification System, we need to Implement WebSockets using Action Cable which is implementing WebSocket. 
Here's the checklist of all the task lists.
- [ ]  Basic Setup of Action Cable.

**Comment by namangupta01** [comment]
@gautamig54, @jywarren  Have a look. I am adding tasks related to Notification Using WebSockets here. Feel free to add.

**Comment by gautamig54** [comment]
@namangupta01 I had a question here. Will we implement a notification system on the website, like a separate page which has all the notifications?

**Comment by namangupta01** [comment]
Yes, we will be going to implement that also. In fact, we should discuss it. Would you like to work on the UI part, design and all of that?

**Comment by namangupta01** [comment]
Can you create an issue for it? So that we along with @jywarren can discuss about it?

**Comment by gautamig54** [comment]
Sure @namangupta01, I would love to work with its UI part. I will open an issue for the same. 
Also, will we create the notification system for the comments only as mentioned in the to-dos by you. And currently, we get mails regarding the comments only. That's why I was thinking of expanding the notification system to adding features like following an user or starring etc. What do you think about this?

**Comment by namangupta01** [comment]
@gautamig54 Opened an issue #5745. Have a look.

**Comment by namangupta01** [comment]
Yeah! That seems interesting. We can add different type of notifications -- as you have mentioned.

**Comment by gautamig54** [comment]
Cool! Let's wait for @jywarren and @gauravano to give their thoughts about this. 
So, should we start with action cable implementation first or this one? According to me, starting with implementing this feature is better if are planning to add different types of notifications. What say @namangupta01 ??

**Comment by namangupta01** [comment]
I think we both can divide our work here and work simultaneously. You working on this feature and me on the web socket integration? And also help out each other in between. What do you say?

**Comment by namangupta01** [comment]
I mean you working on Notification Wall feature? What do you think?

**Comment by gautamig54** [comment]
That sounds great @namangupta01! I will get started with the notification wall!

**Comment by gautamig54** [comment]
Hi @namangupta01, I am starting with implementing the in-app notification system. I was thinking of using activity_notification gem for this purpose. What do you think about this?

**Comment by namangupta01** [comment]
activity_notification seems to be very interesting. Go ahead 👍

**Comment by namangupta01** [comment]
And also we ask @jywarren , @gauravano for the suggestions?

**Comment by gautamig54** [comment]
Sure! Let's wait for @jywarren and @gauravano inputs on this.

**Comment by gauravano** [comment]
Hi @namangupta01 @gautamig54, your work division looks great and clear. Also, activity_notification gem is fine as it's an active project. Thanks!
